src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComEventsMethod.cs

```diff
@@ -58,7 +56,7 @@ public object Invoke(object[] args)
                     {
                         if (_cachedTargetTypes[i] != null)
                         {
-                            args[i] = Enum.ToObject(_cachedTargetTypes[i], args[i]);
+                            args[i] = Enum.ToObject(_cachedTargetTypes[i]!, args[i]!); // TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/34644
```

Comment:
If `args[i]!` is stated shouldn't the type be defined as `object?[]` ?

Reply:
No the problem here is the compiler as you can see in the linked issue as it doesn't track nullability of index'd values. So eventhough we're making sure `_cachedTargetTypes[i]` is not null in the if statement, it doesn't track it propertly. Since `_cachedTargetTypes` is declared as `nullable` the compiler thinks that there is a mismatch here, but not really because we're guaranting it will not be `null` at this point.

Reply:
I am not talking about `_cachedTargetTypes[i]!` which talks about the tracking. The argument `args` itself is declared as `object[]` - which means not null on both sides. If what you are saying is true, then why is `Type?[]?` valid above? I guess I am just confused as to some of the typing.

Reply:
The later ! here isn't necessary. It's leftover from an intermediate state where I'd typed the array as object?[], but then I undid that and neglected to remove the !.

Reply:
Yay, that makes sense. Thank you!

Reply:
It definitely confused as all 😆